posthog/api/test/test_action.py

```diff
@@ -1181,3 +1192,65 @@ def test_breakdown_by_cohort(self):
         ).json()
         self.assertEqual(len(people["results"][0]["people"]), 4)
         self.assertEqual(people["results"][0]["people"][0]["id"], person1.pk)
+
+
+class TestRetention(TransactionBaseTest):
```

Comment:
Hmm. The problem is that it wouldn't be testing the event model directly - it's also testing the data transformations `calculate_retention` are doing which I'd consider worthwhile to test.

The whole code is a bit of a mess organization wise due to trying to fit into `api/action.py`.

What do you think about pulling all of the query building, calculation code, api endpoint code to `api/retention.py` and tests under api/test/test_retention.py?

Reply:
Yep sorry I thought you'd put this in api/event for some reason. I'm a little bit hesitant to add another endpoint to do data analysis, as I like to think about endpoints as mapping pretty concretely to our models. Actions has become the de facto place for analytics endpoints, so I think fine to leave for now.

I do agree it's become a mess but I think the solution is splitting out action.py into sub-endpoints rather than separate endpoints. What do you think?

Reply:
> I do agree it's become a mess but I think the solution is splitting out action.py into sub-endpoints rather than separate endpoints. What do you think?

Sounds good, but what do you have in mind specifically? 

One option would be to create something like `posthog/analytics/retention.py` which would expose all retention related analytics code. We could then do a similar extraction w/ trends for example.

Reply:
I think my suggestion is: let's get this in as is and look at the splitting out later

Reply:
Ok, but what about the original comment regarding this test class?

Reply:
I think that's fine, I thought the entire endpoint was in events.py which is why I was confused, but at least it's consistent now. I do wonder if it's worth separately testing the `query_retention` functionality in `test_event_model.py`, but I'll leave that decision up to you